### PR TITLE
CPM-371: Add context in GetAdditionalProperties* contracts

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml [hunk 1]
```diff
@@ -410,3 +410,8 @@ services:
     Akeneo\Pim\Enrichment\Bundle\Storage\Sql\Category\CountTreesChildren:
         arguments:
             - '@database_connection'
+
+    Akeneo\Pim\Enrichment\Component\Product\Query\PublicApi\GetRawValues:
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml:414]: This is not part of the public API: it's only for internal usage in the Enrichment context. Persistence format should not be exposed outside.

Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml:414]: We have a few places where we need raw values, for indexation for instance (for DQI and now for the table attribute in GE).  
We plug the logic with the `GetAdditionalPropertiesForProductProjectionInterface::fromProductIdentifiers(array $productIdentifiers): array`

We can load the entire products and build our projection using the `getValues()` method, but it's a very long process. For performance we fetch only the raw values and filter them to have only table attribute values. It's way faster.

If we want to keep performance, another solution is to replaced this public API by something like: `GetReadValueCollections::forProductIdentifiersFilteredByAttributeTypes(array $productIdentifiers, array $attributeTypes): array;` that return ReadValueCollection indexed by product identifiers.  

What do you think? Do you have a better idea?

Comment (ahocquard) [reply src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml:414]: Exposing an internal storage format is at the opposite of the philosophy of the hexagonal architecture, and exposing it in PublicAPI is not recommended at all (DQI have some coupling issues so I would not design necessary with that in mind).

We can discuss it all together, the double JSON fetching when indexing bother me much more in term of performance.

Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml]: I push a new version where we put the value collections in a context. This way we don't need to fetch again

**Review by pierallard — APPROVED** [review]

**Review by pierallard — APPROVED** [review]

**Review by mmetayer — COMMENTED** [review]
GTM

**Review by nmarniesse — APPROVED** [review]

**Pull request merged**